src/Parameters/DocumentParametersTrait.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+/*
+ * BigBlueButton open source conferencing system - https://www.bigbluebutton.org/.
+ *
+ * Copyright (c) 2016-2024 BigBlueButton Inc. and by respective authors (see below).
+ *
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation; either version 3.0 of the License, or (at your option) any later
+ * version.
+ *
+ * BigBlueButton is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with BigBlueButton; if not, see <https://www.gnu.org/licenses/>.
+ */
+
+namespace BigBlueButton\Parameters;
+
+use BigBlueButton\Core\Document;
+use BigBlueButton\Core\DocumentFile;
+use BigBlueButton\Core\DocumentUrl;
+use BigBlueButton\Enum\DocumentOption;
+use BigBlueButton\Parameters\Config\DocumentOptions;
+
+trait DocumentParametersTrait
+{
+    /**
+     * @var Document[]
+     */
+    protected array $documents = [];
+
+    public function addDocument(Document $document): self
+    {
+        $this->documents[] = $document;
+
+        return $this;
+    }
+
+    /**
+     * @return Document[]
+     */
+    public function getDocuments(): array
+    {
+        return $this->documents;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function getDocumentsAsXML(): string
+    {
+        $result = '';
+
+        if (!empty($this->documents)) {
+            $xml        = new \SimpleXMLElement('<?xml version="1.0" encoding="UTF-8"?><modules/>');
+            $moduleNode = $xml->addChild('module');
+            $moduleNode->addAttribute('name', 'presentation');
+
+            foreach ($this->documents as $document) {
+                $documentNode = $moduleNode->addChild('document');
+
+                if (null === $documentNode) {
+                    throw new \Exception('XML could not be generated');
+                }
+
+                if ($document->getValidation()) {
+                    if (!$document->isValid()) {
+                        throw new \Exception("Document `{$document->getName()}` is not failed.");
+                    }
+                }
+
+                switch (get_class($document)) {
```

Comment:
Why are we checking for the class name? Isn't preventing extending the library?

Reply:
yes, you are correct. I changed the way things are checked.